tensorboard/plugins/image/summary.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Image summaries and TensorFlow operations to create them.
+
+An image summary stores the width, height, and PNG-encoded data for zero
+or more images in a rank-1 string array: `[w, h, png0, png1, ...]`.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+
+from tensorboard.plugins.image import metadata
+
+
+def op(name,
+       images,
+       max_outputs=3,
+       display_name=None,
+       description=None,
+       collections=None):
+  """Create an image summary op for use in a TensorFlow graph.
+
+  Arguments:
+    name: A unique name for the generated summary node.
+    images: A `Tensor` representing pixel data with shape `[k, w, h, c]`,
+      where `k` is the number of images, `w` and `h` are the width and
```

Comment:
Not in TensorFlow. First three Python files I could think of:
  * https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/variables.py#L144
  * https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/summary/writer/event_file_writer.py#L57
  * https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/summary_ops.py#L40

Also not in TensorBoard:
  * https://github.com/tensorflow/tensorboard/blob/master/tensorboard/plugins/histogram/summary.py#L106 
  * https://github.com/tensorflow/tensorboard/blob/master/tensorboard/backend/event_processing/event_multiplexer.py#L76
  * https://github.com/tensorflow/tensorboard/blob/master/tensorboard/db.py#L58